lib/sha256.c

```diff
@@ -218,9 +221,13 @@ typedef struct sha256_ctx my_sha256_ctx;
 
 static CURLcode my_sha256_init(my_sha256_ctx *ctx)
 {
-  if(CryptAcquireContext(&ctx->hCryptProv, NULL, NULL, PROV_RSA_AES,
-                         CRYPT_VERIFYCONTEXT | CRYPT_SILENT)) {
-    CryptCreateHash(ctx->hCryptProv, CALG_SHA_256, 0, 0, &ctx->hHash);
+  if(!CryptAcquireContext(&ctx->hCryptProv, NULL, NULL, PROV_RSA_AES,
+                         CRYPT_VERIFYCONTEXT | CRYPT_SILENT))
+    return CURLE_OUT_OF_MEMORY;
+
+  if(!CryptCreateHash(ctx->hCryptProv, CALG_SHA_256, 0, 0, &ctx->hHash)) {
+    CryptReleaseContext(ctx->hCryptProv, 0);
```

Comment:
the insert code suggestion button is missing from this pr for some reason

Reply:
Possibly because I made the PR but @danielgustafsson did the commits?

Reply:
> Possibly because I made the PR but @danielgustafsson did the commits?

I see it now. I think what may have happened is I clicked on an individual commit instead of 'Files Changed', because when I did that just now I notice there's no suggestion button.